libs/remix-ui/run-tab/src/lib/run-tab.tsx

```diff
@@ -80,6 +80,22 @@ export function RunTabUI(props: RunTabProps) {
   const [runTab, dispatch] = useReducer(runTabReducer, initialState)
   const REACT_API = { runTab }
   const currentfile = plugin.config.get('currentFile')
+  const [solcVersion, setSolcVersion] = useState<{version: string, canReceive: boolean}>({ version: '', canReceive: true })
+
+  const getVersion = () => {
+    let version = ''
+    try {
+      version = window.location.href.split('=')[5].split('+')[0].split('-')[1].slice(1)
```

Comment:
source of issue was this code. When there is no version in URL params, this throws. Better to check if version is available before splitting again

Reply:
using null coalescing to make sure that there is a default compiler selected. Hope this is agreeable